src/process.py

```diff
@@ -0,0 +1,90 @@
+import os
+import shlex
+import signal
+import subprocess
+import tempfile
+
+#Process(f'xdotool search --all --name {name}').wait().stdout
+
+# with Process(CMD, capture=False):
+        
+#     p = Process('roslaunch test test.launch')
+#     p.wait()
+        
+class Process:
+    """
+    Convenience wrapper for subprocess.Popen. Allows to:
+    - pass cmd as a string despite not using shell
+    - append env variables to spawned process context
+    - capture output (stdout, stderr),
+    - send SIGINT to spawned process group to stop it
+    """
+
+    def __init__(self, cmd, shell=False, env=None, capture=True):
+        self.cmd = cmd
+        self._rc = None
+        self._files = {}
+        kw = dict(shell=shell, close_fds=True)
+        if not shell:
+            cmd = shlex.split(cmd)
+        if env:
+            kw.update(env={**os.environ.copy(), **env})
+        if capture:
+            self._files = dict(
+                stdout=tempfile.NamedTemporaryFile(mode="w", delete=False),
+                stderr=tempfile.NamedTemporaryFile(mode="w", delete=False),
+                stdin=tempfile.NamedTemporaryFile(mode="r", delete=False))
```

Comment:
What if we pass `None` value to `stdin` arg or omit it at all? Not clear why we create an empty temp and read it.

Reply:
just overengineering, if it is working - let's leave it as is.